### PR TITLE
BUG: Let VAR results complete when model has perfect fit

### LINKED ISSUE #7157: VAR error upon retrieving aic
Hello,

I am trying to fit a VAR model on a 45x3 dataframe with a lag of 11 and when I try to retrieve the aic I get the following error

<details>
Traceback (most recent call last):
  File "C:/Projects/test/repro.py", line 8, in <module>
    print('AIC : ', result.aic)
  File "C:\Anaconda3\envs\testenv\lib\site-packages\statsmodels\base\wrapper.py", line 36, in __getattribute__
    obj = getattr(results, attr)
  File "C:\Anaconda3\envs\testenv\lib\site-packages\statsmodels\tsa\vector_ar\var_model.py", line 2092, in aic
    return self.info_criteria['aic']
  File "pandas\_libs\properties.pyx", line 34, in pandas._libs.properties.CachedProperty.__get__
  File "C:\Anaconda3\envs\testenv\lib\site-packages\statsmodels\tsa\vector_ar\var_model.py", line 2073, in info_criteria
    ld = logdet_symm(self.sigma_u_mle)
  File "C:\Anaconda3\envs\testenv\lib\site-packages\statsmodels\tools\linalg.py", line 28, in logdet_symm
    c, _ = linalg.cho_factor(m, lower=True)
  File "C:\Anaconda3\envs\testenv\lib\site-packages\scipy\linalg\decomp_cholesky.py", line 153, in cho_factor
    check_finite=check_finite)
  File "C:\Anaconda3\envs\testenv\lib\site-packages\scipy\linalg\decomp_cholesky.py", line 17, in _cholesky
    a1 = asarray_chkfinite(a) if check_finite else asarray(a)
  File "C:\Anaconda3\envs\testenv\lib\site-packages\numpy\lib\function_base.py", line 499, in asarray_chkfinite
    "array must not contain infs or NaNs")
ValueError: array must not contain infs or NaNs
</details>

I am guessing the reason is that I don't have enough data to train with such a large lag (shorter lags work), but this behavior is a really counterintuitive way of finding out this issue. A better design would be a more informative exception being thrown when trying to call the fit method.

I'm using version 0.11.1

code sample:
```python
import pandas as pd
from statsmodels.tsa.api import VAR

df_train = pd.read_csv('./df_error_lag(11).csv')
model = VAR(df_train.astype(float))
result = model.fit(11)
print('AIC : ', result.aic)
```

test-data:
[df_error_lag(11).zip](https://github.com/statsmodels/statsmodels/files/5553455/df_error_lag.11.zip)

### PR DESCRIPTION
**Pull request by bashtage**
closes #7157

- [X] closes #7157
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/tsa/vector_ar/var_model.py [hunk 1]
```diff
@@ -6,6 +6,8 @@
 ----------
 Lütkepohl (2005) New Introduction to Multiple Time Series Analysis
 """
+from __future__ import annotations
```
Comment (harupy) [inline statsmodels/tsa/vector_ar/var_model.py:9]: Hi @bashtage, `__future__.annotations` is only avaialbe in Python 3.7 or newer. Does this mean `statsmodels` will drop Python 3.6 support?

https://github.com/statsmodels/statsmodels/blob/main/setup.py#L357

Comment (harupy) [reply statsmodels/tsa/vector_ar/var_model.py:9]: I noticed this change in our CI which runs tests using the dev version of statsmodels using Python 3.6:

https://github.com/mlflow/mlflow/runs/3111502285?check_suite_focus=true#step:11:44

Comment (bashtage) [reply statsmodels/tsa/vector_ar/var_model.py:9]: 3.7 is the target for the next release, which is aligned with NumPy and pandas, among others.

Comment (bashtage) [reply statsmodels/tsa/vector_ar/var_model.py:9]: FWIW NumPy is already 2 releases into 3.7 minimum (1.20 and 1.21)

Comment (harupy) [reply statsmodels/tsa/vector_ar/var_model.py:9]: @bashtage Thanks for the quick reply! Got it, we'll update our CI to use Python 3.7.

Comment (harupy) [reply statsmodels/tsa/vector_ar/var_model.py:9]: Are you planning to update this line?

https://github.com/statsmodels/statsmodels/blob/86e1c85e922fe72dd83a4e7869aa0e52807a3dc6/setup.py#L357

Comment (bashtage) [reply statsmodels/tsa/vector_ar/var_model.py:9]: Thanks. in #7590

Comment (harupy) [reply statsmodels/tsa/vector_ar/var_model.py:9]: Awesome, thanks!

**Pull request merged**